src/iexamine.cpp

```diff
@@ -3465,3 +3389,47 @@ iexamine_function iexamine_function_from_string(std::string const &function_name
     debugmsg("Could not find an iexamine function matching '%s'!", function_name.c_str());
     return &iexamine::none;
 }
+
+hack_result iexamine::hack_attempt(player *p) {
```

Comment:
At least this function should take a player by reference, not as pointer. The `iexamine` interface requires a pointer (which should be changed, but requires some overhaul), but this function is not part of the interface and does not need to follow that restriction.